ethers/erc20.nim

```diff
@@ -0,0 +1,52 @@
+import pkg/stint
+import pkg/ethers
+
+export stint
+export ethers
+
+type
+  Erc20Token* = ref object of Contract
+
+  Transfer* = object of Event
+    sender* {.indexed.}: Address
+    receiver* {.indexed.}: Address
+    value*: UInt256
+
+  Approval* = object of Event
+    owner* {.indexed.}: Address
+    spender* {.indexed.}: Address
+    value*: UInt256
+
+method name*(erc20: Erc20Token): string {.base, contract, view.}
```

Comment:
It might be a good idea to explicitly define default behaviour for non-overridden methods in derived classes. Typically, this is either `discard` or `raiseAssert("not implemented")`.

Reply:
That is certainly a good idea in the general case, but here the `{.contract.}` pragma already provides the implementation.